tests/snuba/search/test_backend.py

```diff
@@ -2318,6 +2320,157 @@ def test_compound_message_negation(self):
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(search_filter_query="issue.category:profile my_tag:1")
+        assert list(results) == [self.profile_group_1, self.profile_group_2]
+
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(
+                search_filter_query="issue.type:profile_blocked_thread my_tag:1"
+            )
+        assert list(results) == [self.profile_group_1, self.profile_group_2]
+
+    def test_error_generic_query(self):
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(search_filter_query="my_tag:1")
+        assert list(results) == [
+            self.profile_group_1,
+            self.profile_group_2,
+            self.error_group_2,
+            self.error_group_1,
+        ]
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(
+                search_filter_query="issue.category:[profile, error] my_tag:1"
+            )
+        assert list(results) == [
+            self.profile_group_1,
+            self.profile_group_2,
+            self.error_group_2,
+            self.error_group_1,
+        ]
+
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(
+                search_filter_query="issue.type:[profile_blocked_thread, error] my_tag:1"
+            )
+        assert list(results) == [
+            self.profile_group_1,
+            self.profile_group_2,
+            self.error_group_2,
+            self.error_group_1,
+        ]
+
+    def test_cursor_performance_issues(self):
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(
+                projects=[self.project],
+                search_filter_query="issue.category:profile my_tag:1",
+                sort_by="date",
+                limit=1,
+                count_hits=True,
+            )
+
+        assert list(results) == [self.profile_group_1]
+        assert results.hits == 2
+
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(
+                projects=[self.project],
+                search_filter_query="issue.category:profile my_tag:1",
+                sort_by="date",
+                limit=1,
+                cursor=results.next,
+                count_hits=True,
+            )
+        assert list(results) == [self.profile_group_2]
+        assert results.hits == 2
+
+        with self.feature("organizations:issue-platform"):
+            results = self.make_query(
+                projects=[self.project],
+                search_filter_query="issue.category:profile my_tag:1",
+                sort_by="date",
+                limit=1,
+                cursor=results.next,
+                count_hits=True,
+            )
+        assert list(results) == []
+        assert results.hits == 2
+
+
```

Comment:
can we add a test for the `null` condition where the feature isn't enabled? just to make sure it's all working OK